src/core/Header/HeaderLogo.tsx

```diff
@@ -0,0 +1,65 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import '@itwin/itwinui-css/css/header.css';
+
+import cx from 'classnames';
+import React from 'react';
+
+import { useTheme } from '../utils/hooks/useTheme';
+import { CommonProps } from '../utils/props';
+
+export type HeaderLogoProps = {
+  /**
+   * Logo shown before the main content.
+   */
+  logo: JSX.Element;
+  /**
+   * Click event handler.
+   * Will update the Logo to have mouse and keyboard interaction if provided.
+   */
+  onClick?: () => void;
+  /**
+   * Children is put on the right of the logo.
```

Comment:
Maybe write explicitly that it should be app name?

Reply:
Added